src/header/header.scss

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: $iui-xxs solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
```

Comment:
With 2nd popular screen size (1366 x 768 px) product name is cut:
![image](https://user-images.githubusercontent.com/36186912/116051498-56ab7000-a681-11eb-92cc-a7b9bf48822b.png)
![image](https://user-images.githubusercontent.com/36186912/116051554-6460f580-a681-11eb-9c79-4a2cbc52052f.png)

Reply:
I've made some adjustments so that the left & right don't have their content shrink to fit the center content.
![Screen Shot 2021-04-27 at 4 26 25 PM](https://user-images.githubusercontent.com/849817/116308409-4e743300-a75c-11eb-8015-3513c4b4fb9b.png)

I would expect the dev team to set media-queries for how to handle the search box for example when the screen width gets too narrow.

Reply:
Are you sure this is what we want ?
![z79L7xVv6u](https://user-images.githubusercontent.com/1904889/116309792-32ca5600-a777-11eb-8c0d-23675bf99d9e.gif)

Reply:
Submitted another change where the center content stays more visibly centered, independent of the content in the left & right sections.
![Screen Shot 2021-04-27 at 4 44 28 PM](https://user-images.githubusercontent.com/849817/116310328-b297f680-a75e-11eb-91dd-bf11b17a01e8.png)

Reply:
Actually, I was more focussing on the lower parts where there is no center piece, and the right items get out of view, we can aim for something like below, where we would collapse everything in the right panel to a single (or 2) buttons, but then we need to completely control what's in that area: 
![hM2hKRrnJ6](https://user-images.githubusercontent.com/1904889/116310730-60fc6580-a778-11eb-8cbd-414d3a73c367.gif)